win32ss/user/ntuser/menu.c

```diff
@@ -2992,10 +2992,8 @@ static BOOL FASTCALL MENU_ShowPopup(PWND pwndOwner, PMENU menu, UINT id, UINT fl
     /* We are off the right side of the screen */
     if (x + width > monitor->rcMonitor.right)
     {
-        if ((x - width) < monitor->rcMonitor.left || x >= monitor->rcMonitor.right)
-            x = monitor->rcMonitor.right - width;
-        else
-            x -= width;
+        /* Position menu at right edge of screen */
+        x = monitor->rcMonitor.right - width + 1;
     }
 
     /* We are off the left side of the screen */
```

Comment:
Note:
In this block, it seems to me `|| x >= monitor->rcMonitor.right` "should" be removed, as useless.
But, as `x = 0` was accepted on previous PR, it seems "safer" to keep it for the time being.